bob-common/src/mapper.rs

```diff
@@ -130,19 +171,31 @@ impl Virtual {
         let target_nodes = self.get_target_nodes_for_key(key);
```

Comment:
Please, add a `debug_assert` that checks that all nodes in `target_nodes` are unique. You can do that by collection nodes indices into HashSet and compare lengths

Reply:
Done.